docker/daemon.go

```diff
@@ -360,5 +360,14 @@ func loadDaemonCliConfig(config *daemon.Config, daemonFlags *flag.FlagSet, commo
 		}
 	}
 
+	// ensure that TLS is enabled when TLSVerify is set
+	// in the configuration file
```

Comment:
Maybe we should add the same note here to explain we enable TLS both for tis-verify = true AND false; people may be confused and try to fix it in future. taken from `common.go`;

```
// Regardless of whether the user sets it to true or false, if they
// specify --tlsverify at all then we need to turn on tls
```